src/en/robust-toolbox/ecs.md

```diff
@@ -110,26 +110,25 @@ As you can see, by using composition all our problems from before are solved nea
 - Humans and borgs share the `PlayerControllable` (also known as `Mind` in SS14 code) component, which allows them to be controlled by a player...
 - And most of the game objects above share `Damageable`, which allows them to have "health" and be damaged.
 
-TODO: finish this
+What's wrong with OOP?
 ----
+Object Oriented Programming (OOP) is merely a programming paradigm, or model. In this section, we'll take a very brief view of what motivates our project to dissent from object-oriented data modeling. 
 
-Edit all this text into something proper for a doc... Sigh
+Let's first take a look at inheritance. There are many problems and inflexibility that come with complex inheritance trees. Refer to the tree diagram above, and imagine we had no components. Instead, we have large inheritance tree. In this tree, we have a "machine" base class that has power consumption. We also have a different "mob" base class that has various player-controlled mechanics. Notice that both of these objects are on two separate branches of our component hierarchy. This poses our first design dilemma; what if we want some machines to have mob qualities, or some mobs to have machine qualities? Our system is designed in such a manner where you cannot make a mob have machine qualities or vice versa without making ugly hacks, or a common base for both "machine" and "mob". 
```

Comment:
This duplicates a lot of what is already written starting on line 18. It would be better to work any additions you have into the existing text instead of repeating it.